site/pages/api/create-consequence-operator.test.ts

```diff
@@ -309,4 +318,48 @@ describe("create-consequence-operator API", () => {
             Location: `${CREATE_CONSEQUENCE_OPERATOR_PATH}/${defaultDisruptionId}/${defaultConsequenceIndex}?${returnPath}`,
         });
     });
+
+    it("should redirect to /type-of-consequence when all required inputs are passed and add another consequence is true", async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: bodyData,
+            query: { addAnotherConsequence: "true" },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        await createConsequenceOperator(req, res);
+
+        expect(upsertConsequenceSpy).toHaveBeenCalledTimes(1);
+        expect(upsertConsequenceSpy).toHaveBeenCalledWith(
+            operatorToUpsert,
+            DEFAULT_ORG_ID,
+            mockSession.isOrgStaff,
+            false,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${TYPE_OF_CONSEQUENCE_PAGE_PATH}/${defaultDisruptionId}/1`,
+        });
+    });
+
+    it("should redirect to /type-of-consequence when all required inputs are passed and add another consequence is true and a template", async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: bodyData,
+            query: { addAnotherConsequence: "true", template: "true" },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        await createConsequenceOperator(req, res);
+
+        expect(upsertConsequenceSpy).toHaveBeenCalledTimes(1);
+        expect(upsertConsequenceSpy).toHaveBeenCalledWith(
+            operatorToUpsert,
+            DEFAULT_ORG_ID,
+            mockSession.isOrgStaff,
+            true,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${TYPE_OF_CONSEQUENCE_PAGE_PATH}/${defaultDisruptionId}/1?template=true`,
```

Comment:
Could we write a test case that verifies that the correct index value is passed to the redirection?

Reply:
this does do this as shown by the /1

Reply:
done